common/CMakeLists.txt

```diff
@@ -171,9 +171,19 @@ set(kissfft_srcs
   src/fft/kiss_fftr.c
 )
 
+set(executor_incs
+  include/pcl/experimental/executor/type_trait.h
+  include/pcl/experimental/executor/property.h
+  include/pcl/experimental/executor/executor.h
+)
+
+set(executor_srcs
+  include/pcl/experimental/executor/blocking.cpp
+)
+
 set(LIB_NAME "pcl_${SUBSYS_NAME}")
 include_directories("${CMAKE_CURRENT_SOURCE_DIR}/include")
-PCL_ADD_LIBRARY(${LIB_NAME} COMPONENT ${SUBSYS_NAME} SOURCES ${srcs} ${kissfft_srcs} ${incs} ${common_incs} ${impl_incs} ${tools_incs} ${kissfft_incs} ${common_incs_impl} ${range_image_incs} ${range_image_incs_impl})
+PCL_ADD_LIBRARY(${LIB_NAME} COMPONENT ${SUBSYS_NAME} SOURCES ${srcs} ${kissfft_srcs} ${executor_srcs} ${incs} ${common_incs} ${impl_incs} ${tools_incs} ${kissfft_incs} ${common_incs_impl} ${range_image_incs} ${range_image_incs_impl} ${executor_incs})
```

Comment:
Unrelated to your work, but did anyone wonder why pcl is passing headers into this this command?

Reply:
So that other modules can access the headers by including the module name I assume.

Reply:
these are being registered as target source files. nothing is accessible to downstream targets. targets only propagate include dirs, compile flags, definitions. not files.

Reply:
Maybe ppl messed up the include dirs sometime ago

Reply:
Will investigate this later. Will open an issue on PCL repo to keep track